### DISCUSSION TITLE
COPC support

### DISCUSSION THREAD

**Discussion by RaananHadar** [developmentseed/titiler#504] [Feature Request]
So this is an interesting one. 

COPC a tiled 3d point cloud format that aims to be the COG of point cloud formats. It was recently featured in the latest Cloud Native Geo Spatial outreach event.

Point clouds are not as popular as rasters and I think one of the reasons is the general lack of a good streaming solution. COPC is supported in PDAL so I was wondering how difficult it is to add COPC support for TiTiler.

The only web based alternative i've seen to 3d point clouds is [potree](https://github.com/potree/potree) and I fear that it is unmaintained.

**Comment by vincentsarago** [comment]
Hi @RaananHadar, thanks for starting the discussion.

> to add COPC support for TiTiler.

Can you explain a bit more what you mean? 

TiTiler is a Raster service that mostly aims to create image tiles from raster dataset. For COPC what will be the output format? (3Dtiles?)

I'm not sure we want/need to support COPC directly in TiTiler but we can totally work on something similar for COPC if it doesn't exist yet.

ping @hobu

**Comment by hobu** [comment]
There is a niche for a "titiler" for COPC, but that code does not belong in TiTiler itself.  Overview point clouds of large collections of COPC tiles is indeed a need. If all of the content is spatially organized, that tool could quickly provide overviews of data by subsampling individual tiles. Combined with PDAL, some transformations could be done to the content along the way (like normalizing Intensity or trimming to a polygon). 

TiTiler assumes data are rasters and derived products of TiTiler are also rasters. This doesn't fit point clouds, but the need for an analogous tool is indeed there.

**Reply by jo-chemla** [reply]
There have been discussions revolving around a loaders.gl COPC loader - dedicated [feature request](https://github.com/visgl/loaders.gl/issues/2911) just opened which could be one of the building blocks of such a middleware. 

Related dream [thread](https://twitter.com/jo_chemla/status/1671142245448204291) I wrote a while back on the usefulness of such a Tiled-3D middleware: passing any 3D-tiled datasets (mesh or pointcloud, COPC, OGC 3D-tiles, potree) format into any 3D-tiled supporting application (web via deck.gl, desktop like qgis, unreal engine/unity and any 3D engine where a cesium plugin do exist, esri ecosystem etc) - it might also be useful to start a discussion on this topic on loaders.gl repo itself - edit, just did open that [discussion](https://github.com/visgl/loaders.gl/discussions/2913)
